pkg/operator/controllers/machinehealthcheck/machinehealthcheck_controller.go

```diff
@@ -101,8 +105,13 @@ func (r *Reconciler) Reconcile(ctx context.Context, request ctrl.Request) (ctrl.
 
 // SetupWithManager will manage only our MHC resource with our specific controller name
 func (r *Reconciler) SetupWithManager(mgr ctrl.Manager) error {
+	aroClusterPredicate := predicate.NewPredicateFuncs(func(o client.Object) bool {
+		return strings.EqualFold(arov1alpha1.SingletonClusterName, o.GetName())
+	})
+
 	return ctrl.NewControllerManagedBy(mgr).
-		For(&machinev1beta1.MachineHealthCheck{}).
+		For(&arov1alpha1.Cluster{}, builder.WithPredicates(aroClusterPredicate)).
 		Named(ControllerName).
+		Owns(&machinev1beta1.MachineHealthCheck{}).
```

Comment:
It reconcile it immediately, but after some time it will requeue naturally and it will fix and reconcile it.  

This is the same pattern for geneva logging, MUO, etc.